analyzers/tests/SonarAnalyzer.TestFramework/Verification/Verifier.cs

```diff
@@ -269,11 +269,16 @@ private List<string> PrepareRazorFiles(string projectRoot)
 
     private ProjectBuilder CreateProject(bool concurrentAnalysis)
     {
-        var paths = builder.Paths.Select(TestCasePath).ToArray();
+        var paths = builder.Paths.Select(TestCasePath).GroupBy(IsRazorOrCshtml).ToDictionary(x => x.Key, x => x.ToArray());
+        var razorPaths = paths.GetOrAdd(true, _ => []);
+        var documentPaths = paths.GetOrAdd(false, _ => []);
```

Comment:
I think we can make it a bit more readable:
```suggestion
var allPaths = builder.Paths.Select(TestCasePath).ToList();
var razorPaths = allPaths.Where(IsRazorOrCshtml).ToArray();
var documentPaths = allPaths.Except(razorPaths).ToArray();
```
not sure performance-wise, but should be negligible anyway, we don't get many paths to go through

Reply:
And what about calling them sourceFilePaths and contentFilesPaths? for now they are just razor/cshtml files but who knows in the future. Let me know wdyt